File: src/data/models/CreditCardInvoice.ts
import { DocumentModel } from './DocumentModel';

export class CreditCardInvoice extends DocumentModel {
  constructor(
    public id: string,
    public cardId: string,
    public invoiceDate: Date,
    public year: number,
    public month: number,
    public value: number,
    public paymentDate?: Date,
    public paymentAccountId?: string,
    public paidValue?: number,
    public importInfo?: string
  ) {
    super(id);
  }

  public get name(): string {
    return CreditCardInvoice.makeName(this.year, this.month);
  }

  public static nowName(): string {
    const now = new Date();
    return CreditCardInvoice.makeName(now.getFullYear(), now.getMonth() + 1);
  }

  private static makeName(year: number, month: number): string {
    return year + String(month).padStart(2, '0');
  }
}

File: iOS/Sources/Data/models/CreditCardInvoice.swift
import Foundation

struct CreditCardInvoice: Codable, Identifiable {
    var id: String?
    var cardId: String
    var invoiceDate: Date
    var year: Int
    var month: Int
    var value: Double
    var paymentDate: Date?
    var paymentAccountId: String?
    var paidValue: Double?
    var importInfo: String?

    var name: String {
        String(format: "%04d%02d", year, month)
    }

    var paid: Bool {
        paymentDate != nil && paymentAccountId != nil && paidValue != nil
    }

    static func nowName() -> String {
        let now = Date()
        let calendar = Calendar.current
        let year = calendar.component(.year, from: now)
        let month = calendar.component(.month, from: now)
        return String(format: "%04d%02d", year, month)
    }
}
